apps/api/v1/views.py

```diff
@@ -141,19 +139,51 @@ def users(self, request, pk=None):
                 return Response({"error": "User not found"}, status=status.HTTP_404_NOT_FOUND)
 
 
-class ConfigView(AnsibleBaseDjangoAppApiView, viewsets.ViewSet):
+class SettingView(AnsibleBaseDjangoAppApiView, viewsets.ViewSet):
     permission_classes = [IsAdminUser, AnsibleBaseObjectPermissions]
 
     @extend_schema(operation_id="config_retrieve", description="Get current configuration", responses={200: dict})
     def list(self, request):
-        return Response(DYNACONF.to_dict())
+        setting_dict = DYNACONF.to_dict()
+
+        # Convert any PosixPath objects to strings
+        def serialize_value(value):
+            from pathlib import Path
+
+            if isinstance(value, Path):
+                return str(value)
+            elif isinstance(value, dict):
+                return {k: serialize_value(v) for k, v in value.items()}
+            elif isinstance(value, list):
+                return [serialize_value(item) for item in value]
+            return value
+
+        serializable_config = serialize_value(setting_dict)
+        return Response(serializable_config)
 
     @extend_schema(
         operation_id="config_update", description="Update current configuration", request=dict, responses={204: None}
     )
-    @action(detail=False, methods=["post"])
+    @action(detail=False, methods=["put", "patch"])
     def update_config(self, request):
-        DYNACONF.merge(request.data)
+        """Update the config settings and log the changes."""
+        # Take a snapshot of current settings the change
+        old_settings = {}
+        for key in request.data:
+            old_settings[key] = DYNACONF.get(key)
+
+        # Log each change by calling the helper function
+        for key, new_value in request.data.items():
+            for key, value in request.data.items():
+                DYNACONF.set(key, value)
+            log_setting_change(
+                user=request.user,
+                setting_key=key,
+                new_value=new_value,
+                source="api",
+                request=request,
+            )
+
```

Comment:
We probably want to guard against a user trying to add unrecognized settings key, value pairs.  We could do something like:

```suggestion
        for key, new_value in request.data.items():
             current_value = DYNACONF.get(key)
            if current_value is None:
                return Response(
                    {"error": f"Setting '{key}' does not exist. Cannot create new settings."},
                    status=status.HTTP_400_BAD_REQUEST,
                )

            # Update the setting
            DYNACONF.set(key, new_value)

            # Log the change
            log_setting_change(
                user=request.user,
                setting_key=key,
                new_value=new_value,
                source="api",
                request=request,
            )

```

Reply:
Agreed.